src/ng/directive/form.js

```diff
@@ -268,12 +268,22 @@ FormController.prototype = {
    * @name form.FormController#$setSubmitted
    *
    * @description
-   * Sets the form to its submitted state.
+   * Sets the form to its `$submitted` state. This will also set `$submitted` on all child and
+   * parent forms of the form.
    */
-  $setSubmitted: function() {
-    this.$$animate.addClass(this.$$element, SUBMITTED_CLASS);
-    this.$submitted = true;
-    this.$$parentForm.$setSubmitted();
+  $setSubmitted: function(onlySetOnChildren) {
```

Comment:
The alternative is to create a new private function. Perhaps something like this:

```
$setSubmitted: function() {
  var rootForm = this;
  while(rootForm.$$parentForm) { rootForm = rootForm.$$parentForm; }
  rootForm.$$setSubmitted();
}

$$setSubmitted() {
  this.$$animate.addClass(this.$$element, SUBMITTED_CLASS);
  this.$submitted = true;
  forEach(this.$$controls, function(control) {
    if (control.$$setSubmitted && !control.$submitted) {
      control.$setSubmitted(false);
    }
  });
}
```